apps/site/src/app/(main)/mentor/Form/Form.tsx

```diff
@@ -0,0 +1,167 @@
+"use client";
+
+import { FormEvent, useState } from "react";
+import axios from "axios";
+
+import Textfield from "@/lib/components/forms/Textfield";
+import Button from "@/lib/components/Button/Button";
+
+import AgeInformation from "@/lib/components/forms/shared/AgeInformation";
+import SchoolInformation from "@/lib/components/forms/shared/SchoolInformation";
+import ResumeInformation from "@/lib/components/forms/shared/ResumeInformation";
+import BasicInformation from "./BasicInformation";
+import ProfileInformation from "./ProfileInformation";
+import ShortAnswers from "./ShortAnswers";
+import ExperienceInformation from "./ExperienceInformation";
+
+import hasDeadlinePassed from "@/lib/utils/hasDeadlinePassed";
+import RadioSelect from "@/lib/components/forms/RadioSelect";
+
+import styles from "@/lib/components/forms/shared/Form.module.scss";
+
+const APPLY_PATH = "/api/user/apply";
+const FIELDS_WITH_OTHER = [
+	"pronouns",
+	"ethnicity",
+	"school",
+	"major",
+	"other_mentor_experience",
+];
+const SELECT_OPTIONS = [
+	"frontend",
+	"backend",
+	"mobile",
+	"databases",
+	"ai/ml",
+	"vr",
+	"blockchain",
+	"embedded",
+	"data_science",
+	"cybersecurity",
+];
+
+export default function Form() {
+	const [submitting, setSubmitting] = useState(false);
+	const [sessionExpired, setSessionExpired] = useState(false);
+
+	const handleSubmit = async (
+		event: FormEvent<HTMLFormElement>,
+	): Promise<void> => {
+		// Disable native post submission
+		event.preventDefault();
+
+		// On the chance that a user presses submit after the deadline,
+		// this check is to prevent the application from submitting
+		// and to show the message that applications have closed
+		if (hasDeadlinePassed()) {
+			window.location.reload();
+			return;
+		}
+
+		setSubmitting(true);
+		setSessionExpired(false);
+
+		const formData = new FormData(event.currentTarget);
+
+		// Use other values when selected
+		for (const field of FIELDS_WITH_OTHER) {
+			const otherField = `other_${field}`;
+			if (formData.get(field) === "other") {
+				formData.set(field, formData.get(otherField)!);
+				formData.delete(otherField);
+			}
+		}
+
+		// Compile all technologies the mentor is comfortable with into one array
+		const experienceList: Array<string> = [];
+		for (const fieldName of Array.from(formData.keys())) {
+			if (SELECT_OPTIONS.includes(fieldName)) {
+				experienceList.push(fieldName);
+				formData.delete(fieldName);
+			}
+		}
```

Comment:
Thought: I think if we give all of the checkbox inputs the same name (e.g. `mentor_experienced_technologies` as suggested above) and use explicit values (as we currently are), the name/value pairs will automatically be parsed as an array by FastAPI according to the HTTP specification, (see [MDN Docs for Handling multiple checkboxes](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/checkbox#handling_multiple_checkboxes)). In theory, this means we don't need to do any special handling for the multi-select (except for the "other" option).

Reply:
Fun fact: looks like two years ago, multiple pronouns could be selected: HackAtUCI/HackUCI-Site#27